core-bundle/src/Filesystem/PublicUri/Options.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Filesystem\PublicUri;
+
+final class Options
```

Comment:
Wouldn't those be provder-specific? It looks like at the end of the day it will be some kind of array with unknown options because any provider can accept and support anything. But well, I'm fine having a data object here nevertheless. Maybe we actually can standardize one or two options here which are interesting for multiple providers :) But I think we should already implement setting and getting arbitrary options. Otherwise I cannot pass them on to my own provider (I would need to extend `Options` in order to fulfil the type hint but I cannot because it's final - maybe you're looking for an interface here?).

Reply:
Yes, definitively on my list as well. IMHO it should be an object because you can easily test against features then (`if $options instanceof Foo`, then use them). But not sure whether it should be a base class or an interface. This was only a placeholder for now. :slightly_smiling_face: 

Maybe we should try to get an S3 mount going (not saying that we need to ship it by default) and see what we actually need. Not sure if you can create a free test account there.

Reply:
Can just try locally using a https://min.io container ;)